tactics/tactics.ml

```diff
@@ -1730,9 +1742,10 @@ let general_apply ?(respect_opaque=false) with_delta with_destruct with_evars
           | PretypeError _|RefinerError _|UserError _|Failure _ ->
             Some (try_red_apply red_thm (exn0, info))
           | _ -> None)
-      with Redelimination ->
+      with Redelimination as exn ->
         (* Last chance: if the head is a variable, apply may try
             second order unification *)
+        let exn, info = Exninfo.capture exn in
```

Comment:
This change seems dubious, morally we use the info when reraising the `exn0` which goes with the shadowed `info`

Reply:
Umm, good point, on the other hand if we don't take the stack here we will actually lose the info of taking the `with` branch here. I dunno what to do.

Reply:
Maybe 
```suggestion
      with Redelimination ->
        (* Last chance: if the head is a variable, apply may try
            second order unification *)
        let exn0, info = Exninfo.capture exn0 in
```
and use exn0 in the rest of the branch

Reply:
Umm, isn't this a no-op?